optuna/samplers/nsgaii/_constraints_evaluation.py

```diff
@@ -86,15 +85,45 @@ def _constrained_dominates(
     return violation0 < violation1
 
 
+def _evaluate_penalty(population: Sequence[FrozenTrial]) -> np.ndarray:
+    """Evaluate feasibility of trials in population.
+    Returns:
+        A list of feasibility status T/F/None of trials in population, where T/F means
+        feasible/infeasible and None means that the trial does not have constraint values.
+    """
+
+    penalty: list[float] = []
+    for trial in population:
+        constraints = trial.system_attrs.get(_CONSTRAINTS_KEY)
+        if constraints is None:
+            penalty.append(np.nan)
+        else:
+            assert isinstance(constraints, (list, tuple))
+            penalty.append(sum(v for v in constraints if v > 0))
+    return np.array(penalty)
+
+
 def _validate_constraints(
     population: list[FrozenTrial],
-    constraints_func: Callable[[FrozenTrial], Sequence[float]] | None = None,
+    *,
+    is_constrained: bool = False,
 ) -> None:
-    if constraints_func is None:
+    if not is_constrained:
         return
+    assert len(population) > 0
+
+    num_constraints = None
```

Comment:
Can we compute it in advance?

```suggestion
    num_constraints = max([t.system_attrs.get(_CONSTRAINTS_KEY, []) for t in population], default=0)
```